src/Sulu/Bundle/PageBundle/Resources/js/containers/Form/fields/SettingsVersions.js

```diff
@@ -40,6 +40,75 @@ class PageSettingsVersions extends React.Component<Props> {
         });
     }
 
+    @computed get resourceKey(): string {
+        const {
+            schemaOptions: {
+                resource_key: {
+                    value: resourceKey,
+                } = {},
+            },
+        } = this.props;
+
+        if (resourceKey === undefined || typeof resourceKey !== 'string') {
+            throw new Error(
+                'The "resource_key" schemaOption is mandatory and must be a string, but received ' +
+                typeof resourceKey + '!'
+            );
+        }
+
+        return resourceKey;
+    }
+
+    @computed get listKey(): string {
+        const {
+            schemaOptions: {
+                list_key: {
+                    value: listKey,
+                } = {},
+            },
+        } = this.props;
+
+        if (listKey === undefined || typeof listKey !== 'string') {
+            throw new Error(
```

Comment:
Just an idea: Shouldn't we fallback to the `resourceKey` in case a separate `listKey` is not given? Would make the configuration in XML a bit easier, if all three introduced paramters are the same (which might be the case most of the time).

Reply:
Yeah I like the idea :)